draft-meynell-panrg-scion-research-questions.md

```diff
@@ -201,11 +201,36 @@ This draft assumes the reader is familiar with some of the fundamental concepts
 * One AS belongs to many ISDs?
 
 
+## Beacon Forwarding Policy
+
+The idea behind "beaconing" is to discover all possible paths (loop free and with a fixed maximum length) between two CORE AS.
+Every AS forwards any received beacons to all neighbor CORE ASes unless this would cause a loop or exceed the fixed maximum length.
+Implemented naively, the number of paths (and beacons) grows exponentially with the network size.
+This is currently mitigated, primarily (and efficiently), by forwarding only the five "best PCBs" to neighbours.
+
+This "best PCB" practice has several implications:
+
+* It limits the number of available paths for endpoints such that only five truly diverse paths may be available to a given destination.
```

Comment:
I don't agree with this. It might limit the number of paths if and only if there are no _new_ beacons arriving to the neighboring ASes after **N** beaconing periods. For that to happen, we require that the beaconing step _i+1_ forwards the same beacons than step _i_. OR that the beacons are very short lived and cannot survive longer than one period.
In any other case, period _i+1_ could, and would, forward different beacons to the neighboring ASes.

This is how I would explain it.
- The time needed to propagate all the beacons to the neighboring ASes increments.
- Since all beacons have an expiration time, we have a finite amount of time until a beaconing period will need to forward an already forwarded beacon, just because it expired.
- And, _most importantly_, the semantics of what a "best PCB" means, are defined locally by the **beacon forwarder AS**, not by the receiver AS, who is the actual judge of the quality of the PCB.

Reply:
Sorry, I don't understand what you are saying. Maybe you can clarify?

My assumption is that an AS will rarely change it's what-is-the-best-PCB algorithm. Consequently, they would always forward the same five segments. They may receive new beacons, but as long as the "best 5" are still among them (and I assume they will be, why shouldn'´t they?), they will continue forwarding these 5 PCBs.